Sources/PrimerSDK/Classes/User Interface/OAuth/PrimerWebViewModel.swift

```diff
@@ -9,143 +9,72 @@
     func onRedirect(with url: URL)
-    func onDismiss()
-    func generateWebViewUrl(_ completion: @escaping (Result<String, Error>) -> Void)
-    func getWebViewModel() -> PrimerWebViewModelProtocol
-    func navigate(_ result: Result<Bool, Error>)
-    func tokenize()
+    func onError(_ error: Error)
+    var onCompletion: ((Result<PaymentMethodToken, Error>) -> Void)! { get set }
 }
 
 internal class ApayaWebViewModel: PrimerWebViewModelProtocol {
-
-    var result: Result<Apaya.WebViewResult, ApayaException>?
-    var onCompletion: ((Result<Apaya.WebViewResult, ApayaException>) -> Void)?
+    
+    var onCompletion: ((Result<PaymentMethodToken, Error>) -> Void)!
     
     deinit {
         log(logLevel: .debug, message: "🧨 deinit: \(self) \(Unmanaged.passUnretained(self).toOpaque())")
     }
 
-    private func setResult(_ value: Result<Apaya.WebViewResult, ApayaException>?) {
-        result = value
-    }
-
     func onRedirect(with url: URL) {
-        setResult(Apaya.WebViewResult.create(from: url))
+        let result = Apaya.WebViewResult.create(from: url)
+        
+        switch result {
+        case .success(let result):
+            tokenize(result: result)
+        case .failure(let err):
+            onCompletion?(.failure(err))
+        }
     }
 
-    func onDismiss() {
-        let result = self.result ?? .failure(ApayaException.webViewFlowCancelled)
-        let state: AppStateProtocol = DependencyContainer.resolve()
-        state.setApayaResult(result)
-        onCompletion?(result)
-        setResult(nil)
-    }
-    
-    func reload() {
-        
+    func onError(_ error: Error) {
+        onCompletion?(.failure(error))
     }
     
     func generateWebViewUrl(_ completion: @escaping (Result<String, Error>) -> Void) {
         if configDidLoad() {
             let apayaService: ApayaServiceProtocol = DependencyContainer.resolve()
             apayaService.createPaymentSession(completion)
         } else {
-            // In case we load this view model right away we will be forced to load the config here.
-            // It's probably better if we figure out some middle step so that a payment method view model
-            // never needs to worry about the config having loaded.
-            // This view should never appear without the config.
-            loadConfig { [weak self] result in
+            loadClientToken { [weak self] result in
                 switch result {
                 case .failure(let error):
-                    self?.navigate(.failure(error))
+                    self?.onCompletion?(.failure(error))
                 case .success:
-                    guard let strongSelf = self, strongSelf.configDidLoad() else {
-                        self?.navigate(.failure(PrimerError.configFetchFailed))
+                    guard let self = self, self.configDidLoad() else {
+                        self?.onCompletion?(.failure(PrimerError.configFetchFailed))
```

Comment:
This doesn't work on XCode 12.4
I got a "Variable declared in 'guard' condition is not usable in its body'

Reply:
Oh that's perfectly fine on Xcode > 12.5. I reverted it.